pandas/indexes/multi.py

```diff
@@ -2064,20 +2063,54 @@ def difference(self, other):
             return self
 
         if self.equals(other):
-            return MultiIndex(levels=[[]] * self.nlevels,
-                              labels=[[]] * self.nlevels,
-                              names=result_names, verify_integrity=False)
+            return MultiIndex._create_as_empty(self.nlevels,
+                                               names=result_names)
 
         difference = sorted(set(self._values) - set(other._values))
 
         if len(difference) == 0:
-            return MultiIndex(levels=[[]] * self.nlevels,
-                              labels=[[]] * self.nlevels,
-                              names=result_names, verify_integrity=False)
+            return MultiIndex._create_as_empty(self.nlevels,
+                                               names=result_names)
         else:
             return MultiIndex.from_tuples(difference, sortorder=0,
                                           names=result_names)
 
+    @staticmethod
```

Comment:
actually this should be a method I think, with all parameters optional and overrideable (e.g. by default it would create for nlevels, with the current names)